src/BEditaClient.php

```diff
@@ -357,6 +357,53 @@ public function remove($id) : ?array
         return $this->delete(sprintf('/trash/%s', $id));
     }
 
+    /**
+     * Upload file (POST)
+     *
+     * @param string $filename The file name
+     * @param string $filepath The file full path
+     * @param array $headers Custom request headers
+     * @return array|null Response in array format
+     */
+    public function upload($filename, $filepath, array $headers = ['Content-Type' => '']) : array
+    {
+        $path = sprintf('/streams/upload/%s', $filename);
+        $handle = fopen($filepath, "r");
+        $body = fread($handle, filesize($filepath));
+        fclose($handle);
+        $headers['Content-Type'] = mime_content_type($filepath);
+
+        return $this->post($path, $body, $headers);
+    }
+
+    /**
+     * Create media by type and body data and link it to a stream:
+     *  - `POST /:type` with `$body` as payload, create media object
+     *  - `PATCH /streams/:stream_id/relationships/object` modify stream adding relation to media
+     *  - `GET /:type/:id` get media data
+     *
+     * @param string $streamId The stream identifier
+     * @param string $type The type
+     * @param array $body The body data
+     * @return array|null Response in array format
+     */
+    public function createStreamMedia($streamId, $type, $body) : array
```

Comment:
what about `createMediaFromStream()` or something like that?
or simply `createMedia()`